compiler/src/dotty/tools/dotc/core/Definitions.scala

```diff
@@ -342,17 +340,19 @@ class Definitions {
 
   /** Method representing a throw */
   @tu lazy val throwMethod: TermSymbol = {
-    val argTpe = if (ctx.explicitNulls) OrType(ThrowableType, NullType) else ThrowableType
-    enterMethod(OpsPackageClass, nme.THROWkw, MethodType(List(argTpe), NothingType))
+    enterMethod(OpsPackageClass, nme.THROWkw, MethodType(List(mayNull(ThrowableType)), NothingType))
   }
 
+  // TODO: where ?
+  def mayNull(typ: Type): Type = if (ctx.explicitNulls) OrType(typ, NullType) else typ
```

Comment:
This can probably be moved with the rest of the null-related methods (e.g. with isNullableUnion).
Could you please rename the method to e.g. `maybeNull`?